source/common/http/utility.h

```diff
@@ -158,10 +170,9 @@ Http1Settings parseHttp1Settings(const envoy::api::v2::core::Http1ProtocolOption
  * @param grpc_status the gRPC status code to override the httpToGrpcStatus mapping with.
  * @param is_head_request tells if this is a response to a HEAD request
  */
-void sendLocalReply(bool is_grpc, StreamDecoderFilterCallbacks& callbacks, const bool& is_reset,
-                    Code response_code, absl::string_view body_text,
-                    const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
-                    bool is_head_request);
+void sendLocalReply(const LocalReplyInfo& info, StreamDecoderFilterCallbacks& callbacks,
+                    const bool& is_reset, Code response_code, absl::string_view body_text,
```

Comment:
I'm not sure why previously `is_reset` is passed by reference, but please change to pass by copy.

Reply:
There should be a reason to use the reference type. Below is the comment in the code.
```// encode_headers()) may have changed the referenced is_reset so we need to test it```